test/basic_json.cpp

```diff
@@ -198,7 +198,13 @@ TEST(general, json_stream_operator)
     std::map<std::string, int> m({{"bar", 2}, {"foo", 1}});
     Tree t;
     t.rootref() << m;
-    std::cout << as_json(t) << std::endl;
+    std::stringstream ss;
+    ss << as_json(t);
+    Tree res = c4::yml::parse(to_csubstr(ss.str()));
```

Comment:
don't call `to_csubstr()` on a temporary! substr is a string view, and it makes you responsible for ensuring  that the lifetime of its memory fully contains the lifetime of the substr. 

So setting the csubstr from a temporary is UB and will likely crash. Instead, do this:
````c++
std::string result = ss.str();  // keep the memory in scope!
Tree res = c4::yml::parse(to_csubstr(result));
```

Reply:
Note that the test succeeded because std::string uses the small string optimization, and the produced yaml fits the small buffer. So there was no allocation, and hence no crash. But if the produced YAML is large enough, this will crash.

So I suggest using larger strings in the map and also a couple more nodes to make sure the string is not small and uses allocated memory.

Reply:
I added your suggested changes, however:
 * Even when using longer strings, the test still succeeded, perhaps because the memory was not re-used in the mean time and the read-only access thus still produced the correct result. But are you sure this is a real issue? I don't know enough about c++ to be exactly sure about how the lifetime of object behaves for nested function calls, but I would have expected this to work as long as it all happens within the same line. If you can point me to some explanation, I'd be happy to learn something new.
 * If you really want to avoid people using temporaries as arguments for the `substr`, is there any reason you don't do something like this here? https://stackoverflow.com/questions/11127807/prevent-pass-by-ref-of-temporary-object
But I'm getting ahead of myself I think, let's just take care of this PR :-)

Reply:
That is a nice suggestion, thanks for that. I'm considering add a `to_csubstr(T &&)` overload just raising an error, ideally at compile time.

As for the error, read the comment to understand what's going on.